presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java

```diff
@@ -152,18 +153,27 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
             // return null to throw
             return null;
         }
-        if (name.getSnapshotId().isPresent() && table.snapshot(name.getSnapshotId().get()) == null) {
-            throw new PrestoException(ICEBERG_INVALID_SNAPSHOT_ID, format("Invalid snapshot [%s] for table: %s", name.getSnapshotId().get(), table));
-        }
 
         return new IcebergTableHandle(
                 tableName.getSchemaName(),
                 name.getTableName(),
                 name.getTableType(),
-                Optional.of(name.getSnapshotId().orElse(table.currentSnapshot().snapshotId())),
+                resolveSnapshotId(name, table),
                 TupleDomain.all());
     }
 
+    private Optional<Long> resolveSnapshotId(IcebergTableName name, Table table)
+    {
+        if (name.getSnapshotId().isPresent()) {
+            if (table.snapshot(name.getSnapshotId().get()) == null) {
+                throw new PrestoException(ICEBERG_INVALID_SNAPSHOT_ID, format("Invalid snapshot [%s] for table: %s", name.getSnapshotId().get(), table));
+            }
+            return name.getSnapshotId();
+        }
+        return Optional.ofNullable(table.currentSnapshot())
+                .map(Snapshot::snapshotId);
```

Comment:
nit: Looks like these two lines are short enough to be put into one line. Your call.